### PR TITLE
Automating standard library module dependency graph generation

### PR DESCRIPTION
**Pull request by TharindaDilshan**

### CODE REVIEW

File: release/src/module_dependencies/main.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+				successors.append(i)
+
+			for successor in successors:
+				
+				longestPath = max(nx.all_simple_paths(G, node, successor), key=lambda x: len(x))
+				for n in longestPath[1:-1]:
+					if n in successors:
+						for module in moduleDetailsJSON['modules']:
+							if module['name'] == node:
+								if successor in module['dependents']:
```
Comment (ThisaruGuruge) [inline release/src/module_dependencies/main.py:95]: Can we remove this long `if-else` ladder? Maybe move parts into a separate function, or remove these duplicates in a separate phase? WDYT?

File: release/resources/module_list.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+    "modules": [
+        "module-ballerina-java.arrays",
```
[the remark this answers is not in the document]
Comment (TharindaDilshan) [reply release/resources/module_list.json:15]: Updated

File: .github/workflows/dependency_graph.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                    pip install networkx
+                    python ./release/src/module_dependencies/main.py
+                env:
+                    packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
+                    packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
+            
+            -   name: Commit files
+                id: commit
+                run: |
+                    git config --local user.email "tharindad7@gmail.com"
```
[the remark this answers is not in the document]
Comment (TharindaDilshan) [reply .github/workflows/dependency_graph.yml]: I changed the values to use the stored secrets

File: release/src/module_dependencies/main.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import networkx as nx
+
+# Gets dependencies of ballerina standard library module
+# build.gradle file is accessed through the github search api and decoded to locate the dependencies
+# returns: list of dependencies
+def getDependencies( bal_module ):
+	
+	for files in urllib.request.urlopen(urllib.request.Request("https://api.github.com/repos/ballerina-platform/" + bal_module + "/contents/build.gradle", headers={'Authorization': 'token ' + os.environ['packagePAT']})):
+
+		content = json.loads(files.decode('utf-8'))['content']
```
[the remark this answers is not in the document]
Comment (TharindaDilshan) [reply release/src/module_dependencies/main.py]: Updated

File: release/src/module_dependencies/main.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
+import urllib.request
+import json
+import base64
+import os
+import networkx as nx
+
+# Gets dependencies of ballerina standard library module
+# build.gradle file is accessed through the github search api and decoded to locate the dependencies
+# returns: list of dependencies
+def getDependencies( bal_module ):
```
[the remark this answers is not in the document]
Comment (TharindaDilshan) [reply release/src/module_dependencies/main.py]: Updated

File: release/src/module_dependencies/main.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+	
+	for files in urllib.request.urlopen(urllib.request.Request("https://api.github.com/repos/ballerina-platform/" + bal_module + "/contents/build.gradle", headers={'Authorization': 'token ' + os.environ['packagePAT']})):
```
[the remark this answers is not in the document]
Comment (TharindaDilshan) [reply release/src/module_dependencies/main.py]: Thank you. I think this is better than using the API as there's a limit for API request.

File: release/src/module_dependencies/main.py (hunk 1 above)
Comment (TharindaDilshan) [reply release/src/module_dependencies/main.py]: I moved it to a separate function

**Pull request closed without merging**